src/overlays/actors/ovl_Bg_Ice_Turara/z_bg_ice_turara.c

```diff
@@ -27,23 +41,164 @@ const ActorInit Bg_Ice_Turara_InitVars = {
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgIceTurara_Break(BgIceTurara* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f velvec;
+    Vec3f posvec;
+    s32 phi_s0;
+    s32 sp88;
+
+    Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.posRot.pos, 30, NA_SE_EV_ICE_BROKEN);
+
+    for (sp88 = 0; sp88 != 2; sp88++) {
+        for (phi_s0 = 0; phi_s0 != 10; phi_s0++) {
+
+            posvec.x = this->dyna.actor.posRot.pos.x + Math_Rand_CenteredFloat(8.0f);
+            posvec.y = this->dyna.actor.posRot.pos.y + (Math_Rand_ZeroOne() * arg2) + (sp88 * arg2);
+            posvec.z = this->dyna.actor.posRot.pos.z + Math_Rand_CenteredFloat(8.0f);
+
+            velvec.x = Math_Rand_CenteredFloat(7.0f);
+            velvec.z = Math_Rand_CenteredFloat(7.0f);
+            velvec.y = (Math_Rand_ZeroOne() * 4.0f) + 8.0f;
+
+            EffectSsEnIce_Spawn(globalCtx, &posvec, (Math_Rand_ZeroOne() * 0.2f) + 0.1f, &velvec, D_8089267C,
+                                primColor, envColor, 30);
+        }
+    }
+}
+
+void func_80892220(BgIceTurara* this, GlobalContext* globalCtx) {
+    if ((this->collider.base.acFlags & 2) != 0) {
+        BgIceTurara_Break(this, globalCtx, 50.0f);
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void func_80892280(BgIceTurara* this, GlobalContext* globalCtx) {
+
+    if (this->dyna.actor.xzDistFromLink < 60.0f) {
+        this->shiverTimer = 10;
+        this->actionFunc = BgIceTurara_Shiver;
+    }
+}
+
+void BgIceTurara_Shiver(BgIceTurara* this, GlobalContext* globalCtx) {
+    s16 phi_v0_3;
+    s16 phi_v0_2;
+    f32 sp28;
+
+    if (this->shiverTimer != 0) {
+        this->shiverTimer--;
+    }
+    if (!(this->shiverTimer % 4)) {
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_ICE_SWING);
+    }
+    if (this->shiverTimer == 0) {
+        this->dyna.actor.posRot.pos.x = this->dyna.actor.initPosRot.pos.x;
+        this->dyna.actor.posRot.pos.z = this->dyna.actor.initPosRot.pos.z;
+        Collider_CylinderUpdate(&this->dyna.actor, &this->collider);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->actionFunc = func_80892424;
+        return;
+    }
+    sp28 = Math_Rand_ZeroOne();
+    phi_v0_2 = (Math_Rand_ZeroOne() < 0.5f ? -1 : 1);
+    this->dyna.actor.posRot.pos.x = (phi_v0_2 * ((0.5f * sp28) + 0.5f)) + this->dyna.actor.initPosRot.pos.x;
+    sp28 = Math_Rand_ZeroOne();
+    phi_v0_3 = (Math_Rand_ZeroOne() < 0.5f ? -1 : 1);
+    this->dyna.actor.posRot.pos.z = (phi_v0_3 * ((0.5f * sp28) + 0.5f)) + this->dyna.actor.initPosRot.pos.z;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_80892040.s")
+void func_80892424(BgIceTurara* this, GlobalContext* globalCtx) {
```

Comment:
This function should be written using only if-else, with an early return only after `Actor_Kill`. If you need extra code blocks for matching , use `if(1) {}`.

Reply:
Yes, I should have looked more closely at this after it matched. Thanks